### PR TITLE
Change scim2-filter-parser pin to lower bound

### PR DESCRIPTION
**Pull request by andersk**
There have now been several scim2-filter-parser releases without a corresponding django-scim2 release. We should allow users to benefit from those fixes.

(We could alternatively specify `~=0.3.5` or `>=0.3.5, <0.4` or `>=0.3.5, <1` if you’re more comfortable with that.)

### CODE REVIEW

**Review by logston — APPROVED** [review]
Thanks!

**Comment by logston** [comment]
@caleb15, looks like I've lost permissions on this repo. Can you point me to whoever is managing SCIM things at 15Five these days? We'll need to merge this and then cut a new release.

**Comment by caleb15** [comment]
@logston that would be @powellc . I checked the permission settings and you still have write access. The PR is unmergeable currently because it doesn't meet the merge requirements. We currently require 1 approval and status checks to pass before merging. The CI checks say waiting for status to be reported, so that's why it can't be merged. 

I went over to the actions tab but I didn't see a run for this PR. In https://github.com/15five/django-scim2/settings/actions we currently have it set so first-time contributors (which andersk falls into) require approval to run workflows. There should be a "approve and run" button, but it looks like there is a bug because that's not there :scream: 

https://github.community/t/missing-approve-and-run-button/200572/7

For now I've disabled require status checks to pass before merging. You should be able to merge in now.

**Comment by logston** [comment]
Thanks @caleb15! When I say permissions, I mean the permissions to check the branch settings, etc.

**Comment by logston** [comment]
@powellc, do you mind cutting a new release?

**Comment by powellc** [comment]
@logston happy to. Hope you're well!

**Comment by logston** [comment]
You too! Thanks.

**Comment by powellc** [comment]
Indeed @andersk I don't have access to push to PyPI ... @logston can you do a release?

**Comment by logston** [comment]
Pushed! https://pypi.org/project/django-scim2/0.17.3/

**Pull request merged**
